backend/models/resolutionFactory.js

```diff
@@ -0,0 +1,23 @@
+import config from 'config';
+import { ResolutionInMemoryModel } from './resolutionInMemoryModel.js';
+import { ResolutionRedisModel } from './resolutionRedisModel.js';
+
+class ResolutionFactory {
+  constructor(type) {
+    this.type = type;
+  }
+
+  create() {
+    switch (this.type) {
+      case 'in-memory':
+        return new ResolutionInMemoryModel();
+      case 'redis':
+        return new ResolutionRedisModel();
+    }
+  }
+}
+
+const resolutionFactory = new ResolutionFactory(config.get('db.type'));
```

Comment:
In general, u should initialize and provide config on one lvl upper for all object. It means that current file should not depends on config